source/Engine.cpp

```diff
@@ -2134,6 +2139,37 @@ void Engine::DoWeather(Weather &weather)
 
 
 
+// Determines alternate mouse turning, setting player mouse angle, and right-click firing weapons.
+void Engine::HandleMouseInput(Command &activeCommands)
+{
+	if(activeCommands.Has(Command::MOUSETURNING))
+	{
+		isMouseTurningEnabled = !isMouseTurningEnabled;
+		Preferences::Set("alt-mouse turning", isMouseTurningEnabled);
+	}
+	bool rightMouseButtonHeld = false;
+	int mousePosX;
+	int mousePosY;
+	if((SDL_GetMouseState(&mousePosX, &mousePosY) & SDL_BUTTON_RMASK) != 0)
+		rightMouseButtonHeld = true;
+	double relx = mousePosX - Screen::RawWidth() / 2;
+	double rely = mousePosY - Screen::RawHeight() / 2;
+	Angle mouseAngle = 0;
+	if(relx == 0)
+		mouseAngle = 90;
+	else
+		mouseAngle = (180 / PI) * (atan(rely / relx)) + 90;
```

Comment:
I think it would be good to add a comment somewhere around here to explain whats going on

Reply:
ok will do that

Reply:
Done!